pkgs/development/tools/godot/4/default.nix

```diff
@@ -42,7 +42,7 @@ let
     else "${k}=${builtins.toJSON v}");
 in
 stdenv.mkDerivation rec {
-  pname = "godot";
+  pname = "godot4";
```

Comment:
`pname` should match top-level name. Which is `godot_4`.
The package name and package version separator is `_` in nixpkgs.